pkg/promclient/multi_api.go

```diff
@@ -62,7 +68,7 @@ func NormalizePromError(err error) error {
 type MultiAPIMetricFunc func(i int, api, status string, took float64)
 
 // NewMultiAPI returns a MultiAPI
-func NewMultiAPI(apis []API, antiAffinity model.Time, metricFunc MultiAPIMetricFunc, requiredCount int) *MultiAPI {
+func NewMultiAPI(apis []API, antiAffinity model.Time, metricFunc MultiAPIMetricFunc, requiredCount int, Memory int) *MultiAPI {
```

Comment:
Looking at this again, it seems like this memory limiting functionality might be better suited in a separate client instance (like how the debug one works) -- instead of having to deal with the multiAPI stuff; then we could wire up the single memory watcher at a higher level when we merge the servergroups.